### DISCUSSION TITLE
Unable to achieve scroll animation effect with Link (in Chrome)

### DISCUSSION THREAD

**Discussion by imsahil2001** [vercel/next.js#35010] [Help] [answered]
Started learning next using documentation, but when trying the scroll feature, I was not able to achieve that 

_Here the code's snippet_

` <h2>
        <Link href="/#last-head" >
          <a>Back to Home</a>
        </Link>
      </h2>
`

The element to which I am redirecting is `h1` which is at the home page having the above-mentioned id

Any way to get it working?
Thanks in advance

**Comment by icyJoseph** [comment]
Hi,

This is a thing browsers just do.

Without much code to go by it is hard to answer, but:

1. What happens if you just do use an `a` anchor tag directly?
2. ~~Is any part of the target element visible on the viewport?~~ Is the element already in the viewport? if so, there'll be no scroll.

**Reply by imsahil2001** [reply]
Hi,

I have read your reply, does it mean if the element is on another page altogether it will not work, is that what you saying??
 
Actually, the situation is: 

1.  I am redirecting to an element that is on home page == "/" ,
2. and the path from which I am redirecting is == "/posts/first-post"

**Reply by icyJoseph** [reply]
Aha, I had missed that you linked from another page. 

No, if the element is another page, this should work. I think we'll need to see more code, or understand your setup a little better.

I suspect that maybe your home page DOM is not complete as you arrive to that route? There could be a number of elements at play here. Do you think you can describe a little more?

**Reply by imsahil2001** [reply]
Actually, it doesn't contain much content, it's just a template when we first made the project just that 


So, this is the file structure:
![image](https://user-images.githubusercontent.com/61838801/156763309-99d8b98f-6b7a-4d72-a3d6-45d4c859e1b4.png)

This the last head where i want to go under **index.js**
![image](https://user-images.githubusercontent.com/61838801/156763730-d0cdaad1-3ab5-40cf-b483-d88045baf3c9.png)

here is the code under **first-post.js** which redirects  me to this `h1` of **index.js**:
![image](https://user-images.githubusercontent.com/61838801/156763912-867e6fa7-4f0f-4b77-8515-92bfceb353c6.png)

No doubt it works fine but the scroll effect doesnt happen with it , as under documentation it is mentioned to be _true_ by default

**Reply by icyJoseph** [reply]
Really weird, I've made a quick demo, using the same `FirstPost` component, and using your `last-header` snippet, just under the fold, by placing a div with `120vh` above it, and I see scrolling, like so:

```jsx
import Link from "next/link";

export default function Home() {
  return (
    <>
      <div style={{ height: "120vh" }}>Blank</div>

      <h2 id="last-head">header</h2>

      <Link href="/posts/first-post">
        <a>first post</a>
      </Link>
    </>
  );
}
```

However, I see in Chrome, there's some strange behavior on first load, but both scroll to the anchor as needed. Maybe this document can help you further? https://developer.mozilla.org/en-US/docs/Web/CSS/overflow-anchor/Guide_to_scroll_anchoring#suppression_triggers

Otherwise, I am kinda blank here. Maybe you can record a video of the non-scrolling? or what is it exactly that you are expecting?

**Reply by imsahil2001** [reply]
@icyJoseph I got it Man but partially! The problem was actually in CSS, in Chrome you just have to add  `scroll-behavior: smooth` inside `html` / `body`  in **global.css** file, but even after that, it didn't work back and forth.

This means, if I have two links on different pages each and pointing towards each other's ids' then it didn't work well , the animation got stuck in the middle sometimes and if we refresh then it starts working again 

So, I think it's broken, if you know something, then please add
Thanks for giving your time

**Reply by icyJoseph** [reply, marked as answer]
Right, I see, this has been mixed expectations. 

Scrolling to `some-id` on a url like `https://example.com/#some-id`, has been in browsers for a very long time, and it has not much to do with `scroll-behavior`. This CSS property will make it, well, smooth, as opposed to the sort of immediate scroll downward/upward to the link.

I mean to say that this has nothing to do with Link. The nice thing about using Link is that the entire page doesn't have to navigate to the new URL, instead you get the SPA behavior.

Here's a little video I made showing the behavior, [link](https://webm.red/I4IA).

Strange, but yeah, hopefully you get it working later on.
